base_product_merge/wizard/base_product_merge.py

```diff
@@ -0,0 +1,91 @@
+# Copyright 2023 ForgeFlow S.L. (https://www.forgeflow.com)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+import logging
+
+from openupgradelib import openupgrade_merge_records
+
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+
+_logger = logging.getLogger(__name__)
+
+
+class BaseProductMerge(models.Model):
+    _name = "base.product.merge"
+    _description = "Merges two products"
+
+    @api.model
+    def default_get(self, fields):
+        rec = super().default_get(fields)
+        active_ids = self.env.context.get("active_ids", False)
+        active_model = self.env.context.get("active_model", False)
+        ptype = active_model
+        rec.update({"ptype": ptype})
+        if ptype == "product.product":
+            products = self.env[active_model].browse(active_ids)
+            rec.update({"product_ids": [(6, 0, products.ids)]})
+        else:
+            product_templates = self.env[active_model].browse(active_ids)
+            rec.update({"product_tmpl_ids": [(6, 0, product_templates.ids)]})
+        return rec
+
+    dst_product_id = fields.Many2one("product.product", string="Destination product")
+    product_ids = fields.Many2many(
+        "product.product",
+        "product_rel",
+        "product_merge_id",
+        "product_id",
+        string="Products to merge",
+    )
+    ptype = fields.Selection(
+        [("product.product", "Product"), ("product.template", "Template")]
+    )
+    dst_product_tmpl_id = fields.Many2one(
+        "product.template", string="Destination product"
+    )
+    product_tmpl_ids = fields.Many2many(
+        "product.template",
+        "product_tmpl_rel",
+        "product_tmpl_merge_id",
+        "product_tmpl_id",
+        string="Products to merge",
+    )
+    merge_method = fields.Selection([("sql", "SQL"), ("orm", "ORM")], default="sql")
```

Comment:
```suggestion
    merge_method = fields.Selection([("sql", "SQL"), ("orm", "ORM")], default="sql", required=True)
```

Reply:
@JordiMForgeFlow it is required at view level
https://github.com/OCA/stock-logistics-warehouse/pull/1793/files#diff-896802393fa9cf1986c0180f3d0417d645fab32dac2147933d1f713d0d0190e3R26